server/api_test.go

```diff
@@ -77,66 +74,63 @@ func TestServeFile(t *testing.T) {
 	}()
 
 	assert := assert.New(t)
-	idGen := new(MockPollIDGenerator)
 	api1 := &plugintest.API{}
-	p := &MatterpollPlugin{
-		idGen: idGen,
-	}
+	p := setupTestPlugin(t, api1, samplesiteURL)
 	AllowRequestLogging(api1)
-	p.SetAPI(api1)
 
 	w := httptest.NewRecorder()
-	r := httptest.NewRequest("POST", "/logo_dark.png", nil)
+	r := httptest.NewRequest("POST", fmt.Sprintf("/%s", iconFilename), nil)
 	p.ServeHTTP(nil, w, r)
 
 	result := w.Result()
 
 	bodyBytes, err := ioutil.ReadAll(result.Body)
-	assert.Nil(err)
+	require.Nil(t, err)
```

Comment:
Do you have any policy on which of `assert` and `require` should be used?
I may understand the difference between `assert` and `require`, but I don't know which I should use when asserting.

Reply:
I use `require` when it doesn't make sense to continue a test if a check fails. This is mostly the case when a struct is returned and I want to check its fields. Take https://github.com/matterpoll/matterpoll/blob/55_mux/server/command_test.go#L185 as am example. If `r` is nil the rest of the test will just throw a NPE. Its better to stop the run there and exit. This makes debugging tests easier.

Reply:
Thank you for explaining it. I got it and feel making sense 👍